conans/client/conf/detect.py

```diff
@@ -49,7 +49,7 @@ def _clang_compiler(output, compiler_exe="clang"):
             compiler = "apple-clang"
         elif "clang version" in out:
             compiler = "clang"
-        installed_version = re.search("([0-9]\.[0-9])", out).group()
+        installed_version = re.search("version ([0-9]\.[0-9])", out).group(1)
```

Comment:
This change in the regular expression worries me a little. If some specific clang is not printing the "version" word we could be breaking.  Why did you need it? If there is a good reason I'm ok with it, if don't, better keep it like it was.

Reply:
If it is only for apple-clang I would prefer to limit it to the Apple if to mitigate possible issues

Reply:
But apple-clang was already working, when the compiler is "clang --version". So parsing didn't seem an issue

Reply:
That's a good question, but looking at the output it could be:

```
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin17.4.0
```

Please @pvicente confirm then it is failing and why didn't fail so far.

Reply:
The problem is when using fake gcc in mac os x.

This is the output for gcc:

```
10:31 $ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.0.0 (clang-900.0.38)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
The previous regular expression catches 4.2 as compiler version, see [regex problem](https://regex101.com/r/cT1W9X/4)

When I added version in regex I checked out that clang in linux, clang in apple are returning ``version`` in the output.

MacOSX: 

```
23:21 $ clang --version
Apple LLVM version 9.0.0 (clang-900.0.38)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Linux:

```
conan@8bb2abebb6c8:/Users/pvicente/git/conan/conan$ clang --version
clang version 5.0.0-3 (tags/RELEASE_500/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```